ada/language/ast.py

```diff
@@ -3400,6 +3400,15 @@ def previous_part(go_to_incomplete=(Bool, True)):
             ))).cast(T.BaseTypeDecl)
```

Comment:
Did you profile the perf before/after that ? Seems like replacing an env search by a linear search could have a perf impact ...

Reply:
Also what does this fix? specifically

Reply:
No I haven't, but well functional correctness is more important right? This allows the change in the next commit to work. (When a primitive is declared on an apparent incomplete type, we need to fetch its next_part and attach the primitive to it instead (as IncompleteTypeDecl is not a TypeDecl), which does not work if next_part is based on env lookups as this whole thing is happening during PLE and envs are not fully populated at this point).

Reply:
Yes, functional correctness is definitely more important :) But knowing the perf impact would still be interesting.

Could you run a quick profile on a big project and see what gives ?